### PR TITLE
(BKR-846) re-organized docs

### PR DESCRIPTION
**Pull request by kevpl**
[skip ci]

### CODE REVIEW

**Comment by kevpl** [comment]
rebased due to outside changes.

@puppetlabs/beaker, @er0ck, @cowofevil, ready for a look!

File: README.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # Beaker
 
-Beaker is an acceptance testing harness for Puppet PE and other Puppet Projects.  It can also be used as a virtual machine provisioner - setting up machines, running any configuration on those machines and then exiting.
+Beaker is an acceptance testing harness for Puppet PE and other Puppet Projects.
```
Comment (er0ck) [inline README.md]: i suggest that is a SUT agnostic testing harness focused on interactions across multiple (virtual) machines (its core value add?), VM provisioning, and platform abstraction.

i'd like to see all references to Puppet as a product and Puppet Inc specific internals/infra removed from beaker (other than vmpooler, until the hypervisor stuff is decoupled).  pie in the sky?

File: README.md [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+It can also be used as a virtual machine provisioner - setting up machines,
+running any configuration on those machines and then exiting.
```
Comment (er0ck) [inline README.md]: -configuration- command ?

File: README.md [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
 
-Beaker runs tests written in Ruby with an additional DSL API.  This gives you access to all standard Ruby along with acceptance testing specific commands.
+Beaker runs tests written in Ruby with an additional DSL API.  This gives you
+access to all standard Ruby along with acceptance testing specific commands.
 
 # Installation
```
Comment (er0ck) [inline README.md]: should this now be a section in the larger TOC below?

File: README.md [hunk 4]
```diff
@@ -12,7 +15,18 @@ See [Beaker Installation](docs/Beaker-Installation.md).
 
 Documentation for Beaker can be found in this repository in [the docs/ folder](docs/README.md).
 
-The [Getting Started](docs/How-To-Beaker.md) is a great place to start for new Beaker users.
+A high-level overview of how our docs are organized will help you know where to look for certain things:
```
Comment (er0ck) [inline README.md]: suggest rewording.  maybe just:

`#Table of Contents`

also in docs/README.md

File: README.md [hunk 5]
```diff
@@ -12,7 +15,18 @@ See [Beaker Installation](docs/Beaker-Installation.md).
...
+- [Tutorials](docs/tutorials) take you by the hand through the steps to setup a
+beaker run. Start here if you’re new to Beaker or test development.
+- [Topic guides](docs/topics) discuss key topics and concepts at a fairly high
+level and provide useful background information and explanation.
+- [Rubydocs](http://rubydoc.info/github/puppetlabs/beaker/frames) contains the
+technical reference for APIs and other aspects of Beaker. They describe how it
+works and how to use it but assume that you have a basic understanding of key concepts.
+- [How-to guides](docs/how_to) are recipes. They guide you through the steps
+involved in addressing key problems and use-cases. They are more advanced than
+tutorials and assume some knowledge of how Beaker works.
```
Comment (er0ck) [inline README.md]: (how) do they differ from `Topic guides`?

**Comment by er0ck** [comment]
overall: i like it!

File: README.md (hunk 3 above)
Comment (cowofevil) [inline README.md]: Also the link below is broken when I render the markdown.

File: README.md (hunk 1 above)
Comment (cowofevil) [inline README.md]: +1. Beaker is too closely tied to Puppet, hence the community thinks it is only good for testing Puppet.

File: README.md (hunk 5 above)
Comment (cowofevil) [inline README.md]: Maybe use "concepts" or "overview"?

Comment (kevpl) [inline README.md]: changed to concepts

**Comment by kevpl** [comment]
feedback entered. @er0ck, @cowofevil, @puppetlabs/beaker, this is ready for another look!

File: docs/concepts/argument_processing_and_precedence.md [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-* [Priority of Settings](Argument-Processing-and-Precedence.md#priority-of-settings)
+*  [Environment Variables](argument_processing_and_precedence.md#environment-variables)
+* [Host/Config File Options](argument_processing_and_precedence.md#host-file-options)
+* [ARGV](argument_processing_and_precedence.md#argv-or-provided-arguments-array)
+  * [Supported Command Line Arguments](argument_processing_and_precedence.md#supported-command-line-arguments)
+* [Options File Values](argument_processing_and_precedence.md#options-file-values)
+  * [Example Options File](argument_processing_and_precedence.md#example-options-file)
+* [Default Values](argument_processing_and_precedence.md#default-values)
+  * [Beaker Default Values](argument_processing_and_precedence.md#beaker-default-values)
+* [Priority of Settings](argument_processing_and_precedence.md#priority-of-settings)
```
Comment (er0ck) [inline docs/concepts/argument_processing_and_precedence.md]: i think these anchor links can be relative to current page (no need to update for filename)?

ie:
##headers create implicit #headers anchor

File: docs/tutorials/creating_a_test_environment.md [hunk 7]
```diff
@@ -1,4 +1,7 @@
-Hosts/Nodes/SUTs are defined in the --hosts (--config) file in Yaml format.  This file defines each node in the test configuration. The file can be saved anywhere and used with `beaker --hosts yourhost.cfg` see [The Command Line](The-Command-Line.md) for more info.
+Hosts/Nodes/SUTs are defined in the --hosts (--config) file in Yaml format.
+This file defines each node in the test configuration. The file can be saved
+anywhere and used with `beaker --hosts yourhost.cfg`
```
Comment (er0ck) [inline docs/tutorials/creating_a_test_environment.md]: let's be uniform about the filename extension we use for host configs.  beaker outputs them as .yaml (and in certain cases .yml :-\).  perhaps this should say `yourhost.yaml`?

**Comment by kevpl** [comment]
@er0ck feedback entered, thanks for the quick 👀 !

I don't think we're ready for a standardized footer yet. I think that there's a story beginning to come out of the tutorials folder that will lend itself to that experience though. I'd like to wait for the conclusion of @pinkypie's [BKR-860](https://tickets.puppetlabs.com/browse/BKR-860) work (internal-only link), and base it on that, however.

**Comment by er0ck** [comment]
vagrant rake task. yes!  that works for me.
i think we should also ticket to remove all puppet specific stuff from the main parts of the tutorials and focus the tutorials on generic, multi-host interactions.  maybe leave the single howto on modules testing, and expand it.  
thoughts?

File: README.md [hunk 8]
```diff
@@ -1,39 +1,60 @@
...
-The [Getting Started](docs/How-To-Beaker.md) is a great place to start for new Beaker users.
+## Table of Contents
 
-#Beaker Libraries
-
-Beaker functionality has been extended through the use of libraries available as gems. See the
-[complete list](docs/Beaker-Libraries.md) for available gems. See the [beaker-template documentation](https://github.com/puppetlabs/beaker-template/blob/master/README.md) for documentation on creating beaker-libraries.
+- [Tutorials](docs/tutorials) take you by the hand through the steps to setup a
+beaker run. Start here if you’re new to Beaker or test development.
+- [Concepts](docs/concepts) discuss key topics and concepts at a fairly high
```
Comment (er0ck) [inline README.md]: use the word 'reference' here?

**Comment by kevpl** [comment]
@cowofevil? @tvpartytonight?

**Comment by kevpl** [comment]
rebased to resolve conflicts. @tvpartytonight merge party?

File: README.md [hunk 9]
```diff
@@ -1,39 +1,60 @@
 # Beaker
 
-Beaker is an acceptance testing harness for Puppet PE and other Puppet Projects.  It can also be used as a virtual machine provisioner - setting up machines, running any configuration on those machines and then exiting.
+Beaker is a test harness focused on acceptance testing via interactions
+between multiple (virtual) machines. It provides platform abstraction between
+different Systems Under Test (SUTs), and it can also be used as a virtual machine
+provisioner - setting up machines, running any commands on those machines
+and then exiting.
```
Comment (tvpartytonight) [inline README.md]: I'm into the [oxford comma](https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/), are you?

File: README.md [hunk 10]
```diff
@@ -1,39 +1,60 @@
...
 
-Beaker runs tests written in Ruby with an additional DSL API.  This gives you access to all standard Ruby along with acceptance testing specific commands.
+Beaker runs tests written in Ruby, and provides additional DSL methods.  This
```
Comment (tvpartytonight) [inline README.md]: Maybe define `DSL`, in case people mistake it for [this](https://en.wikipedia.org/wiki/Digital_subscriber_line)?

**Comment by tvpartytonight** [comment]
LGTM! 👍
